src/microbe_stage/MicrobeHUD.cs

```diff
@@ -348,6 +354,70 @@ protected override void UpdateAbilitiesHotBar()
         unbindAllHotkey.Pressed = Input.IsActionPressed(unbindAllHotkey.ActionName);
     }
 
+    protected override void UpdateHoverInfo(float delta)
+    {
+        stage!.HoverInfo.Process(delta);
+
+        // Show hovered compound information in GUI
+        foreach (var compound in hoveredCompoundControls)
+        {
+            var compoundControl = compound.Value;
+            stage.HoverInfo.HoveredCompounds.TryGetValue(compound.Key, out float amount);
+
+            // It is not useful to show trace amounts of a compound, so those are skipped
+            if (amount < Constants.COMPOUND_DENSITY_CATEGORY_VERY_LITTLE)
+            {
+                compoundControl.Visible = false;
+                continue;
+            }
+
+            compoundControl.SetText(compound.Key.Name);
+            compoundControl.SetDescription(GetCompoundDensityCategory(amount) ?? TranslationServer.Translate("N_A"));
+            compoundControl.SetDescriptionColor(GetCompoundDensityCategoryColor(amount));
+            compoundControl.Visible = true;
+        }
+
+        // Refresh list
+        mouseHoverPanel.ClearEntries("species");
+        mouseHoverPanel.ClearEntries("chunk");
+
+        // Show the entity's name and count of hovered entities
+        hoveredEntities.Clear();
+
+        foreach (var entity in stage.HoverInfo.InspectableEntities)
+        {
+            var category = string.Empty;
+
+            if (entity is Microbe microbe)
+            {
+                if (microbe.IsPlayerMicrobe)
+                {
+                    // Special handling for player
+                    var label = mouseHoverPanel.AddItem("species", entity.InspectableName);
+                    label?.SetDescription(TranslationServer.Translate("PLAYER"));
+                    continue;
+                }
+
+                category = "species";
+            }
+            else if (entity is FloatingChunk)
+            {
+                category = "chunk";
+            }
+
+            hoveredEntities.TryGetValue((category, entity.InspectableName), out int count);
+            hoveredEntities[(category, entity.InspectableName)] = count + 1;
```

Comment:
The tuple on these two lines could be a separate variable to not create the tuple twice.